_posts/2017-03-07-hddtools.md

```diff
@@ -0,0 +1,323 @@
+
+
+```r
+UK <- getData("GADM", country = "GBR", level = 0)
+plot(UK, col = NA, border = "darkgrey")
+
+# Data60UK full catalogue
+allData60UK <- catalogueData60UK()
+hydroALL <- SpatialPointsDataFrame(coords = allData60UK[, c("Longitude",
+                                                         "Latitude")], 
+                                data = allData60UK)
+plot(hydroALL, add = TRUE, col = "blue", pch = 1)
+```
+
+![](/assets/blog-images/2013-03-07-hddtools/unnamed-chunk-12-1.png)<!-- -->
+
+```r
+head(allData60UK)
+```
+
+```
+##   stationID        River          Location gridReference Latitude
+## 1     22001       Coquet           Morwick      NU234044 55.33310
+## 2     22006        Blyth   Hartford Bridge      NZ242800 55.11381
+## 3     23006   South Tyne      Featherstone      NY672610 54.94257
+## 4     24004 Bedburn Beck           Bedburn      NZ117321 54.68382
+## 5     25005        Leven      Leven Bridge      NZ444120 54.50139
+## 6     25006        Greta Rutherford Bridge      NZ033122 54.50511
+##   Longitude
+## 1 -1.632691
+## 2 -1.622163
+## 3 -2.513539
+## 4 -1.820046
+## 5 -1.315911
+## 6 -1.950550
+```
+
+Given the station identification number (in the column stationID), time series of areal precipitation and streamflow discharge can be downloaded using the function `tsData60UK()`. In the example below I show how to get the time series for the first station in the table, for a given temporal window.
+
+
+```r
+# Extract time series for the first station
+stationID <- catalogueData60UK()$stationID[1]
+
+# Extract time series for a specified temporal window
+twindow <- seq(as.Date("1988-01-01"), as.Date("1989-12-31"), by = "days")
+MorwickTSplot <- tsData60UK(stationID = stationID, 
+                            plotOption = TRUE,
+                            twindow = twindow)
+```
+
+![](/assets/blog-images/2013-03-07-hddtools/unnamed-chunk-13-1.png)<!-- -->
+
+#### MOPEX
+This source contains historical hydrometeorological data and river basin characteristics for hundreds of river basins and from a range of climates in the US. As with the previous dataset, hddtools contains functions to dowload the catalogue and time series. The example below shows how to download the MOPEX catalogue and the time series for the first station in the catalogue.
+
+
+```r
+# MOPEX full catalogue
+allMOPEX <- catalogueMOPEX()
+
+# Extract time series 
+BroadRiver <- tsMOPEX(stationID = allMOPEX$stationID[1], 
+                      plotOption = TRUE)
+```
+
+![](/assets/blog-images/2013-03-07-hddtools/unnamed-chunk-14-1.png)<!-- -->
+
+# Future developments
+The hddtools complements R's growing list of packages in environmental web technologies. The package contains functions to interact with a limited number of sources and there is a long list of useful datasets that could be potentially added, e.g. the real-time precipitation data from the UK MetOffice.
```

Comment:
To encourage people to contribute, would you consider opening an issue for adding real-time precipitation data from the UK MetOffice?
Do you have other issues you might open, for example to work on documentation?